### ISSUE TITLE
Stacklayout does not layout  properly.

### ISSUE THREAD

**Issue by RoyiNamir** [NativeScript/nativescript-angular#1195]
I have this simple layout markup : 
```
<StackLayout  orientation="horizontal" #myStack>

		<StackLayout  width="300" backgroundColor="red">
		</StackLayout>

		<StackLayout width="300" backgroundColor="green">
		</StackLayout>

</StackLayout>
 ```

As you can see , both are same width  of 300: 

Let's see this : 

![image](https://user-images.githubusercontent.com/2842123/36017565-f17878c8-0d80-11e8-8d32-cd1b5d299608.png)

Notice that green is also 300 but we only see a part of it  since  `300+300 > screen size`.

Ok so let's try to animate ` #myStack` (!!)  to the left , in order to see the left green : 

```
  ngOnInit(): void {
        setTimeout(() => {
               this.myStack.nativeElement.animate({
                translate: { x: -300, y: 0 },
                duration: 2000,
                curve: enums.AnimationCurve.easeIn
            });
         },1000)
   }
```

![image](https://i.imgur.com/1gNfQfh.gif)

??

What is this white area on the right ? there should be  also a green section there

Basically  this is the situation : 

![image](https://user-images.githubusercontent.com/2842123/36017851-15dc662e-0d82-11e8-9201-c45212b323b1.png)

So i'm expecting the green from the right to be scrolled to the left , i'm basically trying to move `#myStack` left.

Bug ?

[PLAYGROUND](https://play.nativescript.org/?template=play-ng&id=vWxZ04)

**Comment by NickIliev** [comment]
@RoyiNamir this is expected behavior.
What is happening is that by default the root layout will have an effective width (and height) as the screen size.
You need to explicitly create the wider container if you want to have an effective width larger than the screen width.

There are several approaches on how to achieve that.
- use container with fixed width - [demo Playground here](https://play.nativescript.org/?template=play-tsc&id=98hwAY)
```
<GridLayout rows="auto, *" columns="600" backgroundColor="lightgray">
	<Button row="0" text="animate" (tap)="animate()"></Button>
		<StackLayout row="1" orientation="horizontal" #myStack>
			<StackLayout width="300" backgroundColor="red">
				<Label text="Label"></Label>
			</StackLayout>

			<StackLayout width="300" backgroundColor="green">
				<Label text="Label"></Label>
			</StackLayout>
		</StackLayout>
</GridLayout>
```
Note that our container GridLayout has `columns` set to 600.

- Another option is instead of creating fixed size container to use ScrollView (which will measure its children so the children should have predefined size - in your case width="300") - [demo Playground here](https://play.nativescript.org/?template=play-ng&id=vWxZ04&v=3)
```
<GridLayout rows="auto, *" backgroundColor="lightgray">
	<Button row="0" text="animate" (tap)="animate()"></Button>
	<ScrollView row="1" orientation="horizontal"> <!-- another option is using ScrollView-->
		<StackLayout orientation="horizontal" #myStack>
			<StackLayout width="300" backgroundColor="red">
				<Label text="Label"></Label>
			</StackLayout>

			<StackLayout width="300" backgroundColor="green">
				<Label text="Label"></Label>
			</StackLayout>
		</StackLayout>
	</ScrollView>
</GridLayout>
```
In the example above the container element is not needed (used just to create the animate Button).
The `ScrollView` will measure its children (300 + 300 = 600 width) and will take the space needed.

**Comment by RoyiNamir** [comment]
@NickIliev NO words to thank you.

**Comment by RoyiNamir** [comment]
@NickIliev This seems to work with root gridview.
Why ain't it working with root stacklayout ? 

```
<StackLayout  width="600" backgroundColor="gray">
 	<StackLayout  orientation="horizontal" width="600" #myStack>
		<StackLayout width="300" backgroundColor="red">
			<Label text="Label"></Label>
		</StackLayout>

		<StackLayout width="300" backgroundColor="green">
			<Label text="Label"></Label>
		</StackLayout>
	</StackLayout>

</StackLayout>
  ```

**Comment by NickIliev** [comment]
@RoyiNamir I have investigated further and it seems that the default behavior for layouts with a larger width is to be clipped to fit the screen width (height).
The only thing that would make a layout to render itself with larger width/height than the current screen is when its size (or the size of any nested children) is set to **infinity**.  That would prevent the explicit clipping as there is no mechanism to clip from infinity. Of course, we can not directly set our width/height to infinity but in the source code, this is done when we are setting **auto** for Grid's rows and columns or when we use ScrollView.

**Comment by RoyiNamir** [comment]
@NickIliev  But if you look at your solution : 
```
<GridLayout rows="auto, *" columns="600" backgroundColor="lightgray">
```

We're not setting `columns="auto"` , but 600.

So how does it fit with : "_this is done when we are setting `auto` for Grid's rows and `columns` _"

**Comment by NickIliev** [comment]
@RoyiNamir `AbsoluteLayout` is also measured with infinity so it sholud also do the trick!
